spec/controllers/api/manifests_controller_spec.rb

```diff
@@ -0,0 +1,87 @@
+require "rails_helper"
+
+RSpec.describe Api::ManifestsController, type: :controller do
+  describe "GET #show" do
+    render_views
+
+    context "in production environments" do
+      environments = Dir
+        .glob("config/deploy/*.rb")
+        .map { |file| Pathname.new(file).basename(".rb").to_s }
+
+      environments.each do |env|
+        it "return 200 for #{env}" do
+          original_env = ENV["SIMPLE_SERVER_ENV"]
+          ENV["SIMPLE_SERVER_ENV"] = env
+          allow(File).to receive(:read).with("public/manifest/#{env}.json").and_call_original
+
+          expect(File).to receive(:read).with("public/manifest/#{env}.json")
+
+          get :show
+
+          expect(response).to be_ok
+          expect(response.body).to eq(File.read("public/manifest/#{env}.json"))
+
+          ENV["SIMPLE_SERVER_ENV"] = original_env
+        end
+      end
+
+      it "return 404 for an unknown env" do
+        original_env = ENV["SIMPLE_SERVER_ENV"]
+        ENV["SIMPLE_SERVER_ENV"] = "unknown"
+
+        get :show
+
+        expect(response).to be_not_found
+
+        ENV["SIMPLE_SERVER_ENV"] = original_env
+      end
+    end
+
+    context "in non-production environments" do
+      environments = %w[development review]
+
+      environments.each do |env|
+        it "returns a dynamic manifest for #{env}" do
+          original_env = ENV["SIMPLE_SERVER_ENV"]
+          original_host = ENV["SIMPLE_SERVER_HOST"]
+          original_protocol = ENV["SIMPLE_SERVER_HOST_PROTOCOL"]
+
+          ENV["SIMPLE_SERVER_ENV"] = env
+          ENV["SIMPLE_SERVER_HOST"] = "simple.example.com"
+          ENV["SIMPLE_SERVER_HOST_PROTOCOL"] = "https"
```

Comment:
Would it be simpler to stub `ENV` for these specs? Something like [this thing](https://github.com/simpledotorg/simple-server/blob/bce5cbd14cd13d562ff4974e668c309eb47185d6/spec/jobs/appointment_notification/missed_visit_job_spec.rb#L22).

Reply:
Ah, it is fewer lines of code, and guaranteed to be contained to the scope of the spec. I'll rework it.